docs/CInterop.md

```diff
@@ -0,0 +1,94 @@
+# C Interop
+
+This is an extension of what is was covered in the [Language Guide](./LanguageGuide.md#c-interop).
+
+## Managed types
+
+In Carp types like `String` and `Array` are _managed_ types in that they are
+allocated on the Heap and the compiler will automatically free the allocated
+memory when they go out of scope. We'll see how we can go from these managed
+type to C and back.
+
+### String
+
+To use a managed `String` with a C function requiring a `char*` you can use the
+`String.cstr` function that will turn your `(Ref String)` into `(Ptr CChar)`:
+
+
+```clojure
+(register puts (Fn [(Ptr CChar)] ()))
+
+(let [a-str @"A string."]
+  (puts (String.cstr &a-str)))
+(puts (String.cstr "Hello"))
+```
+
+Might be a good idea to not exposed to the end-user the C type directly:
```

Comment:
Might I suggest the edit:

`It might be a good idea not to expose the C type to the end user directly:`

Reply:
I think that sentence is relatively easily misread. Possible alternatives:

* `You may want to hide the C type from the end-user`
* `It might not be a good idea to expose the C type to the end user directly`